From 2f34ecf5f233528f3bed479cd4d8b3cfdff0c582 Mon Sep 17 00:00:00 2001
From: builduser <builduser@archlinux>
Date: Thu, 25 Dec 2025 15:22:36 +0100
Subject: [PATCH] Fix build with Python 3.14

SWIG 4.4.0 uses multi-phase module initialization for Python 3.14+, placing
%init code inside SWIG_mod_exec() which returns int, not a pointer. Return -1
instead of nullptr to indicate initialization error.
---
 bindings/python/sigrok/core/classes.i | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/bindings/python/sigrok/core/classes.i b/bindings/python/sigrok/core/classes.i
index 97898405..9963e721 100644
--- a/bindings/python/sigrok/core/classes.i
+++ b/bindings/python/sigrok/core/classes.i
@@ -84,7 +84,9 @@ typedef guint pyg_flags_type;
      */
     if (!GLib) {
         fprintf(stderr, "Import of gi.repository.GLib failed.\n");
-#if PY_VERSION_HEX >= 0x03000000
+#if PY_VERSION_HEX >= 0x030e0000
+        return -1;
+#elif PY_VERSION_HEX >= 0x03000000
         return nullptr;
 #else
         return;
-- 
2.52.0

